lua/nvim-web-devicons.lua

```diff
@@ -1079,11 +1079,32 @@ local function set_icon(user_icons)
   end
 end
 
+local function get_icon_color(name, ext, opts)
+  if not loaded then
+    setup()
+  end
+
+  local icon_data = icons[name]
+
+  if not icon_data then
+    icon_data = icons[ext]
+
+    if not icon_data and ((opts and opts.default) or global_opts.default) then
+      icon_data = default_icon
+    end
+  end
+
+  if icon_data then
+    return icon_data.icon, icon_data.color
+  end
+end
```

Comment:
this function could be way more small:
```lua
local has_default = (opts and opts.default) or global_opts.default
local icon_data = icons[name] or icons[ext] or (has_default and default_icon)
if icon_data then return icon_data.icon, icon_data.color
```

Reply:
I followed the existing implementation, and I think get_icon can be improved and implemented in the same way.

Reply:
I've fixed both, so check them out.